docs/4.4/aws-oss-guide.md

```diff
@@ -185,42 +190,65 @@ An example policy is shown below:
     "Version": "2012-10-17",
     "Statement": [
         {
-            "Sid": "AllAPIActionsOnTeleportClusterState",
-            "Effect": "Allow",
-            "Action": "dynamodb:*",
-            "Resource": "arn:aws:dynamodb:REGION:ACCOUNTID:table/DYNAMODB-RESOURCE-A"
-        },
-        {
-            "Sid": "AllAPIActionsOnTeleportClusterStateStreams",
-            "Effect": "Allow",
-            "Action": "dynamodb:*",
-            "Resource": "arn:aws:dynamodb:REGION:ACCOUNTID:table/DYNAMODB-RESOURCE-A/stream/*"
-        },
-        {
-            "Sid": "AllAPIActionsOnTeleportClusterEvents",
+            "Sid": "ClusterStateStorage",
             "Effect": "Allow",
-            "Action": "dynamodb:*",
-            "Resource": "arn:aws:dynamodb:REGION:ACCOUNTID:table/DYNAMODB-RESOURCE-B"
+            "Action": [
+                "dynamodb:BatchWriteItem",
+                "dynamodb:UpdateTimeToLive",
+                "dynamodb:PutItem",
+                "dynamodb:DeleteItem",
+                "dynamodb:Scan",
+                "dynamodb:Query",
+                "dynamodb:DescribeStream",
+                "dynamodb:UpdateItem",
+                "dynamodb:DescribeTimeToLive",
+                "dynamodb:CreateTable",
+                "dynamodb:DescribeTable",
+                "dynamodb:GetShardIterator",
+                "dynamodb:GetItem",
+                "dynamodb:UpdateTable",
+                "dynamodb:GetRecords"
+            ],
+            "Resource": [
+                "arn:aws:dynamodb:<region>:<account_id>:table/<cluster_state_table_name>",
+                "arn:aws:dynamodb:<region>:<account_id>:table/<cluster_state_table_name>/stream/*"
+            ]
         },
         {
-            "Sid": "AllAPIActionsOnTeleportClusterEventsStreams",
+            "Sid": "ClusterEventsStorage",
             "Effect": "Allow",
-            "Action": "dynamodb:*",
-            "Resource": "arn:aws:dynamodb:REGION:ACCOUNTID:table/DYNAMODB-RESOURCE-B/stream/*"
+            "Action": [
+                "dynamodb:CreateTable",
+                "dynamodb:BatchWriteItem",
+                "dynamodb:UpdateTimeToLive",
+                "dynamodb:PutItem",
+                "dynamodb:DescribeTable",
+                "dynamodb:DeleteItem",
+                "dynamodb:GetItem",
+                "dynamodb:Scan",
+                "dynamodb:Query",
+                "dynamodb:UpdateItem",
+                "dynamodb:DescribeTimeToLive",
+                "dynamodb:UpdateTable"
+            ],
+            "Resource": [
+                "arn:aws:dynamodb:<region>:<account_id>:table/<events_table_name>",
+                "arn:aws:dynamodb:<region>:<account_id>:table/<events_table_name>/index/*"
+            ]
         },
         {
```

Comment:
I _think_ that we only use a locks table in our reference Terraform - we only need `cluster_state_table_name` and `events_table_name` for regular Teleport setups. You can delete this third section.

Reply:
@travelton Do you remember why we had these locks, I'm pretty sure I've not had to do it myself.

Reply:
I'm pretty sure the only use of the lock table is here: https://github.com/gravitational/teleport/blob/11a82524bdb7a6a7059955818ccb97f1ea407e5d/assets/aws/files/bin/teleport-lock#L25

We can just remove this.

Reply:
Agreed. I believe this IAM role is specific to the HA Terraform example.

Reply:
Fixed.